### PR TITLE
CS-1373: Handlebars upgrade to fix security vulnerabilities

### PR DESCRIPTION
**Pull request by asaxena-sugarcrm**

### CODE REVIEW

**Review by nportik-sugarcrm — APPROVED** [review]
I like the changes made, great job!

**Review by ewahlquist-sugarcrm — APPROVED** [review]

**Review by rogeryang-sugarcrm — APPROVED** [review]
Looks good to me

File: lib/handlebars/compiler/javascript-compiler.js [hunk 1]
```diff
@@ -11,29 +11,45 @@ JavaScriptCompiler.prototype = {
...
-    } else if (JavaScriptCompiler.isValidJavaScriptVariableName(name)) {
-      ret = parent + "." + name;
-    }
-    else {
-      ret = parent + "['" + name + "']";
+    // Block the above dangerous properties altogether from being used. If they are used in a template, we assume it
+    // could be to exploit the lib since the keywords don't make sense for actual template compilation or rendering
+    // unlike 'constructor' which could be someone's occupation :) lol
+    // See https://snyk.io/vuln/SNYK-JS-HANDLEBARS-534988
+    if (dangerousProperties.includes(name)) {
```
Comment (ewahlquist-sugarcrm) [inline lib/handlebars/compiler/javascript-compiler.js]: `includes` won't work in IE, is this our way of officially saying goodbye to IE support? 😄

Comment (asaxena-sugarcrm) [reply lib/handlebars/compiler/javascript-compiler.js]: haha, as per Bill, we stopped support for IE as of 10.3. 😆

**Review by RobertGonzalez — APPROVED** [review]

**Pull request merged**